geopandas/tools/sjoin.py

```diff
@@ -186,6 +180,351 @@ def find_contains(a1, a2):
             columns={"_key_left": "_key_right", "_key_right": "_key_left"}
         )
 
+    # ------------------- HANDLE HOW PARAM, CREATE FINAL DF --------------------
+    return _join_results(
+        result,
+        left_df,
+        right_df,
+        how,
+        left_index_name,
+        right_index_name,
+        index_left,
+        index_right,
+        lsuffix,
+        rsuffix,
+    )
+
+
+def sjoin_nearest(
+    left_df,
+    right_df,
+    how="inner",
+    lsuffix="left",
+    rsuffix="right",
+    search_radius=None,
+    max_search_neighbors=None,
```

Comment:
If we set both to None by default, that it will have very poor performance. I would say we should set a sensible default to use spatial index without the need of setting any of these manually.

Reply:
I thought about this. For neighbors, the sensible default is `max_search_neighbors=1`, but then we might make the issue we were talking about in the previous comments more frequent. For radius, I'm just no sure what a sensible default would be (we also don't know what units are being used), but I am open to suggestions.

Reply:
I would keep radius as None as it is data specific and set `max_search_neighbors` to safer value then 1 (3? we should test it)

Reply:
I'm happy to go with any number. Unfortunately, I think `max_search_neighbors` is also dataset-specific, so I am hesitant to make any recommendations myself. A number too large won't provide much speedup.

Reply:
As a counterpoint to `many users will use default and then look for details if something won't work`, I feel like if users try the function and it is too slow for their needs, the first thing they will do is read the docs, so as long as it's clear there, the end result should be the same. If I personally had to pick, I'd rather the default be slow than unexpected results.

Reply:
Yeah, but there is slow as using something like 10 for the max neighbours and slow with None. The difference would be in the orders of magnitude. I would use nearest in my work matching 200k polygons to 60k lines, so even super safe default of 100 would save a lot of time.

Reply:
Sounds good. if you have time, could you benchmark against your dataset with a couple of different values to get a better idea of what a good default might be? I only tested with the earth lowres dataset.

Reply:
I'll try.